### ISSUE TITLE
feat(tabs): add support for ng-show and hide/show layout directives

### ISSUE THREAD

**Issue by Chidan** [angular/material#4456]
Hi there,

hide attribute  and ng-show doesnt work on <md-tab>, below is what I am trying to do.
<md-tab hide label="Admin" ui-sref="admin" md-active="navController.$state.is('admin')"></md-tab>
![image](https://cloud.githubusercontent.com/assets/5826495/9663983/825cae4a-5267-11e5-80f7-e4e222f75db6.png)

OR 
<md-tab ng-show="navController.$root.currentUser" label="Admin" ui-sref="admin" md-active="navController.$state.is('admin')"></md-tab>
![image](https://cloud.githubusercontent.com/assets/5826495/9664005/9cb44b90-5267-11e5-9b16-2bb2e69f7903.png)

both dont hide the tab, is there any way to conditional show a tab?

Regards,
Chidan

**Comment by paulflo150** [comment]
Can use ng-if: 

````
<md-tab data-ng-if="ctrl.showTab" label="Tab2">Tab2 content</md-tab>
````

http://codepen.io/anon/pen/RWNZyM

**Comment by Chidan** [comment]
@paulflo150  thanks for example, yes ng-if works but ng-show, and hide attributes are not working, is this a bug or by design?

**Comment by opadillaintegrify** [comment]
+1  Using ng-if together with $mdMedia to simulate hide attribute as a workaround

**Comment by jangidgirish** [comment]
when `ng-if` use in `<md-tab>` then `<md-tabs>` **md-selected**  not working, is it bug or work another way?

**Comment by mustela** [comment]
Having the same issue, `ng-show/hide` won't be supported?

**Comment by matt-neill** [comment]
+1

**Comment by yevkbb** [comment]
+1

**Comment by ganeshghalame** [comment]
+1

**Comment by ThomasBurleson** [comment]
This issue is closed as part of our ‘Surge Focus on Material 2' efforts.
      For details, see our forum posting @ http://bit.ly/1UhZyWs.

**Comment by Skripatch** [comment]
The issue is still present.  And ng-if doesn't suit my case. I have ng-controllers inside tabs content and have to init them just once.

**Comment by Beny-Muler** [comment]
+1 the issue is still present.

**Comment by smithscripts** [comment]
This worked for me. You have to modify the source...

In the MdTab directive at appox. line: 25119
Add  "hide:     '=?ngHide'" to the isolated scope
    
```
scope:    {
      active:   '=?mdActive',
      disabled: '=?ngDisabled',
      hide:     '=?ngHide',
      select:   '&?mdOnSelect',
      deselect: '&?mdOnDeselect'
}
```

then in the MdTabItem directive at appox. line 25183

Add this watch
```
function MdTabItem () {
  return {
    require: '^?mdTabs',
    link:    function link (scope, element, attr, ctrl) {
      if (!ctrl) return;
      ctrl.attachRipple(scope, element);

      scope.$watch('tab.scope.hide', function (value) {
          if (value)
              element.css('display', 'none');
          else
              element.css('display', 'block');
      });
    }
  };
}
```

**Comment by Avishaidev** [comment]
@smithscripts wroks perfect !!! Thanks

**Comment by xzessmedia** [comment]
its so annoying, so many controls are just unperfect and broken in some ways... people should think before using this library in serious projects

**Comment by tkegan** [comment]
Just ran into a related bug where controls on an md-tab with the ng-if directive and no default value are not bound properly. Tested Angular 1.5.7 (don't ask...) and 1.7.2. My solution was to use the ng-disabled directive and a bit of css:

`.md-tab.md-disabled { display:none; }`

Not helpful if you use ng-disabled as intended of course.

**Comment by Splaktar** [comment]
I have re-opened this issue to indicate that we would welcome a community contribution to add this functionality to the tabs component.

**Comment by BaldemarJanrain** [comment]
@Splaktar there's already a contribution to this, @smithscripts did it https://github.com/angular/material/issues/4456#issuecomment-296236790

**Comment by Splaktar** [comment]
@BaldemarJanrain right, I see that, but we need a PR for it, not just a comment. Also instead of using a watch in all cases, we would use a `MutationObserver` for improved performance ([example here](https://github.com/angular/material/blob/v1.1.10/src/components/navBar/navBar.js#L482-L507)).

If you would like to submit a PR, I would be happy to help you with it.
